libPhoneNumber/NBPhoneNumberUtil.m

```diff
@@ -907,8 +908,7 @@ - (NSString *)normalizeHelper:(NSString *)sourceString
   NSMutableString *normalizedNumber = [[NSMutableString alloc] init];
 
   for (NSUInteger i = 0; i < numberLength; ++i) {
-    unichar character = [sourceString characterAtIndex:i];
-    NSString *charString = [NSString stringWithCharacters:&character length:1];
+    NSString *charString = [sourceString substringWithRange:NSMakeRange(i, 1)];
     NSString *newDigit = [normalizationReplacements objectForKey:[charString uppercaseString]];
```

Comment:
If you wanted some more performance how about taking the uppercaseString outside the loop and have

NSString *upperString = [sourceString uppercaseString]; and then iterate through upperString.

Reply:
I did some benchmark, it looks like moving uppercaseString up doesn't help much, because we will need to do another substring call from non-uppercased string (line 916) so we don't really get a performance gain.